grammar.js

```diff
@@ -853,13 +853,31 @@ module.exports = grammar({
       ')',
     ),
 
+
+
+    _default_expression: $ => seq(
+        $.keyword_default,
+            choice(
+            seq(
+                '(',
+                    $._inner_default_expression,
+                ')',
+            ),
+            $._inner_default_expression,
+        )
+    ),
+    _inner_default_expression: $ => choice(
+        $.literal,
+        $.invocation,
+    ),
```

Comment:
default expressions are a slightly bigger subset of `_expression` than this -- you can set `default 1+1`, cast with `default (5/2)::int`, or if you have a composite type representing an (int, text) tuple `default (1, 'tag')`. Many of the possibilities don't usually make a ton of sense but they _are_ legal :)

Out of `_expression` I think we need:

- $.literal
- $.list (for composite types)
- $.case (legal, kind of nonsensical)
- $.cast
- alias($.implicit_cast, $.cast)
- $.invocation
- $.binary_expression
- $.unary_expression
- $.array

The binary & unary expressions are kind of escape hatches into illegal behavior since they include all of `_expression` but I'm not sure there's any practical way around that.

Reply:
Ah SQL... *sigh*

Alright. I will just add all of these

Thanks for the research @dmfay